CMakeLists.txt

```diff
@@ -25,25 +25,27 @@ option(ACTS_BUILD_EVERYTHING "Build with most options enabled (except HepMC3 and
 # core related options
 set(ACTS_PARAMETER_DEFINITIONS_HEADER "" CACHE FILEPATH "Use a different (track) parameter definitions header")
 option(ACTS_FORCE_ASSERTIONS "Force assertions regardless of build type" OFF)
+# external library options
+option(ACTS_USE_SYSTEM_LIBS "Use system libraries by default" OFF)
 # plugins related options
 option(ACTS_BUILD_PLUGIN_AUTODIFF "Build the autodiff plugin" OFF)
-option(ACTS_USE_SYSTEM_AUTODIFF "Use autodiff provided by the system instead of the bundled version" OFF)
+option(ACTS_USE_SYSTEM_AUTODIFF "Use autodiff provided by the system instead of the bundled version" ${ACTS_USE_SYSTEM_LIBS})
```

Comment:
I wonder if for example `ACTS_USE_SYSTEM_AUTODIFF` gets updated if `ACTS_USE_SYSTEM_LIBS` gets changed in a later `cmake` call.

For `ACTS_BUILD_EVERYTHING` we use a different mechanism for example

Reply:
Yeah sadly I don't think that is something that CMake can handle, because it's rubbish. Indeed `ACTS_BUILD_EVERYTHING` works slightly differently, but that _forces_ everything to be build, rather than setting the default to build everything (but allowing the user to still turn off individual components) if I remember correctly.